planner/cascades/testdata/transformation_rules_suite_out.json

```diff
@@ -219,14 +219,141 @@
           "Group#0 Schema:[Column#14]",
           "    Projection_5 input:[Group#1], Column#13",
           "Group#1 Schema:[Column#13,test.t.a]",
-          "    TopN_8 input:[Group#2], test.t.a:asc, offset:2, count:1",
-          "Group#2 Schema:[Column#13,test.t.a]",
-          "    Projection_2 input:[Group#3], plus(test.t.a, test.t.b)->Column#13, test.t.a",
+          "    Projection_2 input:[Group#2], plus(test.t.a, test.t.b)->Column#13, test.t.a",
+          "Group#2 Schema:[test.t.a,test.t.b]",
+          "    TopN_9 input:[Group#3], test.t.a:asc, offset:2, count:1",
           "Group#3 Schema:[test.t.a,test.t.b]",
           "    TiKVSingleGather_7 input:[Group#4], table:t",
           "Group#4 Schema:[test.t.a,test.t.b]",
           "    TableScan_6 table:t, pk col:test.t.a"
         ]
+      },
+      {
+        "SQL": "select c from t order by t.a limit 1",
+        "Result": [
+          "Group#0 Schema:[test.t.c]",
+          "    Projection_5 input:[Group#1], test.t.c",
+          "Group#1 Schema:[test.t.c,test.t.a]",
+          "    Projection_2 input:[Group#2], test.t.c, test.t.a",
+          "Group#2 Schema:[test.t.a,test.t.c]",
+          "    TopN_11 input:[Group#3], test.t.a:asc, offset:0, count:1",
+          "Group#3 Schema:[test.t.a,test.t.c]",
+          "    TiKVSingleGather_7 input:[Group#4], table:t",
+          "    TiKVSingleGather_9 input:[Group#5], table:t, index:c_d_e",
+          "Group#4 Schema:[test.t.a,test.t.c]",
+          "    TableScan_6 table:t, pk col:test.t.a",
+          "Group#5 Schema:[test.t.a,test.t.c]",
+          "    IndexScan_8 table:t, index:c, d, e"
+        ]
+      },
+      {
+        "SQL": "select c from t order by t.a + t.b limit 1",
+        "Result": [
+          "Group#0 Schema:[test.t.c]",
+          "    Projection_5 input:[Group#1], test.t.c",
+          "Group#1 Schema:[test.t.c,test.t.a,test.t.b]",
+          "    Projection_2 input:[Group#2], test.t.c, test.t.a, test.t.b",
+          "Group#2 Schema:[test.t.a,test.t.b,test.t.c]",
+          "    TopN_9 input:[Group#3], plus(test.t.a, test.t.b):asc, offset:0, count:1",
+          "Group#3 Schema:[test.t.a,test.t.b,test.t.c]",
+          "    TiKVSingleGather_7 input:[Group#4], table:t",
+          "Group#4 Schema:[test.t.a,test.t.b,test.t.c]",
+          "    TableScan_6 table:t, pk col:test.t.a"
+        ]
+      },
+      {
+        "SQL": "select 1 as a from dual limit 1",
+        "Result": [
+          "Group#0 Schema:[Column#1]",
+          "    Limit_3 input:[Group#1], offset:0, count:1",
+          "Group#1 Schema:[Column#1]",
+          "    Projection_2 input:[Group#2], 1->Column#1",
+          "Group#2 Schema:[]",
+          "    TableDual_1 rowcount:1"
+        ]
```

Comment:
This test case is not useful. It won't generate topN.

Reply:
I'll remove it.